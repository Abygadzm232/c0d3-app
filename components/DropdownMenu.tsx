import React from 'react'
import DropdownButton from 'react-bootstrap/DropdownButton'
import Dropdown from 'react-bootstrap/Dropdown'

//a null item indicates a dropdown divider
export type Item = {
  title: string
  path?: string
  as?: 'a' | 'button'
  onClick?: Function
} | null

type DropDownMenuProps = {
  drop?: 'down' | 'right' | 'up' | 'left'
  items: Item[]
  title: string
  size?: 'sm' | 'lg' | undefined
  variant?:
    | 'primary'
    | 'secondary'
    | 'success'
    | 'info'
    | 'warning'
    | 'danger'
    | 'none'
}

export const DropdownMenu: React.FC<DropDownMenuProps> = ({
  drop = 'down',
  variant = 'none',
  title,
  size,
  items
}) => {
  const menuItems = items.map((item: Item, itemsIndex: number) =>
    !item ? (
      <Dropdown.Divider key={itemsIndex} />
    ) : (
      <Dropdown.Item
        as={item.as || 'a'}
        key={itemsIndex}
        href={item.path}
        onClick={() => item.onClick && item.onClick(item.title)}
      >
        {item.title}
      </Dropdown.Item>
    )
  )

  return (
    <DropdownButton title={title} variant={variant} size={size} drop={drop}>
      {menuItems}
    </DropdownButton>
  )
}
