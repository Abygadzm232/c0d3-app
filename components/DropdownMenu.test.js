import React from 'react'
import { render } from '@testing-library/react'
import { DropdownMenu } from './DropdownMenu'

const dropdownMenuItems = [
  { title: 'Lessons', path: '/admin/lessons' },
  null,
  { title: 'Users', path: '/admin/users' },
  { title: 'Alerts', path: '/admin/alerts' }
]

let testBtnOnClick = ''

describe('MdInput Component', () => {
  test('Should render divider when an item is null', () => {
    const { container, queryByRole } = render(
      <DropdownMenu title="Admin" items={dropdownMenuItems} />
    )
    const div = queryByRole('separator')

    expect(div).toBeTruthy
    expect(container).toMatchSnapshot()
  })

  test('Should change value of testBtnOnClick upon click', () => {
    dropdownMenuItems[0].onClick = () => (testBtnOnClick = 'lolztestsucceeded')
    const { container, queryByText } = render(
      <DropdownMenu title="Admin" items={dropdownMenuItems} />
    )
    const btn = queryByText('Lessons')

    expect(testBtnOnClick).toEqual('lolztestsucceeded')
    expect(container).toMatchSnapshot()
  })
})
