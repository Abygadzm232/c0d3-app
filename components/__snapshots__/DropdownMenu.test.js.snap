// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MdInput Component Should change value of testBtnOnClick upon click 1`] = `
<div>
  <div
    class="dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown-toggle btn btn-none"
      type="button"
    >
      Admin
    </button>
    <div
      aria-labelledby=""
      class="dropdown-menu"
      style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none; margin: 0px;"
      x-placement="bottom-start"
    >
      <button
        class="dropdown-item"
        href="/admin/lessons"
      >
        Lessons
      </button>
      <div
        class="dropdown-divider"
        role="separator"
      />
      <button
        class="dropdown-item"
        href="/admin/users"
      >
        Users
      </button>
      <button
        class="dropdown-item"
        href="/admin/alerts"
      >
        Alerts
      </button>
    </div>
  </div>
</div>
`;

exports[`MdInput Component Should render divider when an item is null 1`] = `
<div>
  <div
    class="dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown-toggle btn btn-none"
      type="button"
    >
      Admin
    </button>
  </div>
</div>
`;
